pkg/kubectl/cmd/replace.go

```diff
@@ -71,61 +99,102 @@ func NewCmdReplace(f cmdutil.Factory, out io.Writer) *cobra.Command {
 		Example: replaceExample,
 		Run: func(cmd *cobra.Command, args []string) {
 			cmdutil.CheckErr(cmdutil.ValidateOutputArgs(cmd))
-			err := RunReplace(f, out, cmd, args, options)
-			cmdutil.CheckErr(err)
+			cmdutil.CheckErr(options.Complete(f, cmd))
+			cmdutil.CheckErr(options.Validate(cmd))
+			cmdutil.CheckErr(options.Run(f, args))
```

Comment:
Don't pass factory here, extract whatever you need in `Complete` and use that in `Run`. The same applies to `args`.

Reply:
We need the factory here, as it's used to obtain a resource builder.
Args are also used by a resource builder in the `o.forceReplace` method.

Reply:
Builder can be part of options, similarly to https://github.com/kubernetes/kubernetes/blob/ee4d90aaa61150139cdcd67a73e22da8cb226dc6/pkg/kubectl/cmd/create/create_job.go#L58